src/main/kotlin/com/zepben/evolve/database/sqlite/cim/upgrade/changesets/ChangeSet55.kt

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.database.sqlite.cim.upgrade.changesets
+
+import com.zepben.evolve.database.paths.DatabaseType
+import com.zepben.evolve.database.sqlite.cim.upgrade.Change
+import com.zepben.evolve.database.sqlite.cim.upgrade.ChangeSet
+
+internal fun changeSet55() = ChangeSet(
+    55,
+    listOf(
+        // Network Change
+        `Add phase code columns for usage_points`,
+        `Create table curve_data`,
+        `Create table grounding_impedances`,
+        `Create table petersen_coils`,
+        `Create table reactive_capability_curves`,
+        `Create table synchronous_machines`,
+        `Create table synchronous_machine_reactive_capability_curves`,
+    )
+)
+
+// ###################
+// # Network Changes #
+// ###################
+
+@Suppress("ObjectPropertyName")
+private val `Add phase code columns for usage_points` = Change(
+    listOf(
+        "ALTER TABLE usage_points ADD COLUMN phase_code TEXT NULL;"
+    ),
+    targetDatabases = setOf(DatabaseType.NETWORK_MODEL)
+)
+
+@Suppress("ObjectPropertyName")
+private val `Create table curve_data` = Change(
+    listOf(
+        """CREATE TABLE curve_data (
+            curve_mrid TEXT NOT NULL,
+            x_value NUMBER NOT NULL,
+            y1_value NUMBER NOT NULL,
+            y2_value NUMBER NULL,
+            y3_value NUMBER NULL
+        );""".trimIndent(),
+        "CREATE UNIQUE INDEX curve_data_curve_mrid_x_value ON curve_data (curve_mrid, x_value);",
+        "CREATE INDEX curve_data_curve_mrid ON curve_data (curve_mrid);"
+    ),
+    targetDatabases = setOf(DatabaseType.NETWORK_MODEL)
+)
+
+@Suppress("ObjectPropertyName")
+private val `Create table grounding_impedances` = Change(
+    listOf(
+        """CREATE TABLE grounding_impedances (
```

Comment:
todo: The table names here are correct, but they are incorrect in the actual table definitions. This means we do not have a test in place in the upgrade runner that makes sure the upgraded database can actually be loaded. 

There should be a test that check all changesets can be run. Can you please add a final check to that test to ensure the database readers can be run over the database. There is no need to add any data to check, a blank set of tables will still check the SQL matches.

Reply:
I thought that's the upgraderunner, which does have the newest change set in there.

Reply:
and if it's the changeset test it does have the changeset55 in there too

Reply:
There are tests for running all the change sets in `ChangeSetTest` that makes sure all the change sets can be run. What it is clearly not doing correctly is checking that the database that comes out the end of that is actually valid from the perspective of the database readers/writers, it is just making sure the change sets run.

This could be seen in your code before you updated the names as the table class was creating a table named `grounding_impedance`, but this changeset was creating it called `grounding_impedances`. If you revert the table class to using `grounding_impedance`, all tests will still pass, but if you upgrade an old database and then try and load it into EWB (or the routes test server), it will fail with missing table errors.

The test that is running each of the change sets (`test change sets`) needs to be expanded to make sure it can load the post-split upgraded database (without any records) via the database reader to confirm the schema matches.

Reply:
Resolved by ensuring we can prepare the SQL statements for all tables against the upgraded database in the change set validation test.